syntax = "proto3";

package proto;

option cc_enable_arenas = true;

service Greeting {
    rpc GreetPing (GreetRequest) returns (GreetResponse) {};
    rpc AskName (AskNameRequest) returns (AskNameResponse) {};
}

message GreetRequest {
    string name = 1;
    string greet_msg = 2;
}

message GreetResponse {
    string greet_msg = 1;
}

message AskNameRequest {
    string ask_num = 1;
}

message AskNameResponse {
    string name = 1;
}
